galley/pkg/server/server.go

```diff
@@ -84,10 +86,18 @@ func defaultPatchTable() patchTable {
 
 // New returns a new instance of a Server.
 func New(a *Args) (*Server, error) {
-	return newServer(a, defaultPatchTable())
+	var convertK8SService bool
```

Comment:
I think this needs to be part of arguments. New() is also used in tests, so having this environment check will cause problems with integration testing. 

This should be handled the same way the command-line flags are being handled.